docs/groupKey.js

```diff
@@ -0,0 +1,19 @@
+function groupKey(groupingProperties, instance) {
```

Comment:
Do we have a unit test for this function?

Should we have executable code in our `docs` folder, or rather a [Jupyter notebook](https://github.com/n-riesco/ijavascript) with text and code and results intermingled, literate-programming style, following Donald E. Knuth's example?

Reply:
The code is not meant to be executed but serves documentary purposes only. And github.com offers nice features like history and edit-by-pull-request, would Jupyter also offer this?

Reply:
I've nevertheless added a test case, and it immediately spotted superfluous code.